metrics-operator/controllers/metrics/controller.go

```diff
@@ -126,6 +115,43 @@ func (r *KeptnMetricReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return reconcile, err
 }
 
+func (r *KeptnMetricReconciler) updateMetric(metric *metricsapi.KeptnMetric, value string, rawValue []byte, reconcile ctrl.Result, err error) ctrl.Result {
+	if metric.Spec.Range != nil && metric.Spec.Range.StoredResults > 0 {
+		intervalResult := metricsapi.IntervalResult{
+			LastUpdated: metav1.Time{Time: time.Now().UTC()},
+			ErrMsg:      "",
+		}
+
+		if err != nil {
+			r.Log.Error(err, "Failed to evaluate the query", "Response from provider was:", (string)(rawValue))
+			intervalResult.ErrMsg = err.Error()
+		} else {
+			intervalResult.Value = value
+			intervalResult.Range = metric.Spec.Range
+		}
+
+		if len(metric.Status.IntervalResults) >= int(metric.Spec.Range.StoredResults) {
+			metric.Status.IntervalResults = append(metric.Status.IntervalResults[1:], intervalResult)
+		} else {
+			metric.Status.IntervalResults = append(metric.Status.IntervalResults, intervalResult)
+		}
+	} else {
+		if err != nil {
+			r.Log.Error(err, "Failed to evaluate the query", "Response from provider was:", (string)(rawValue))
+			reconcile = ctrl.Result{Requeue: false}
```

Comment:
i think this line can be removed as we should also requeue in case there is an error - e.g. when there is temporarily no data for a metric this should not prevent any further reconciliation of the KeptnMetric

Reply:
Fixed

Reply:
The failing unit test seems to be occurring due to the removal of this line @bacherfl
Should this line really be removed, as according to logic in the main branch, we don't requeue if there's an error over here. https://github.com/keptn/lifecycle-toolkit/blob/main/metrics-operator/controllers/metrics/controller.go#L113

Reply:
true, in this case let's adapt the assertion in the unit test

Reply:
Do you mean to switch the `want` field from `controllerruntime.Result{Requeue: false, RequeueAfter: 0}` to `controllerruntime.Result{Requeue: true, RequeueAfter: 10 * time.Second}` for the failing tests?

Reply:
yes, adapting the expected result in the test should do the trick

Reply:
fixed